deface/deface.py

```diff
@@ -102,6 +128,106 @@ def cam_read_iter(reader):
         yield reader.get_next_data()
 
 
+def has_overlap(det, other):
+    # In openCV the coordinator start at the top left corner of the image
+    x1, y1, x2, y2, _ = det
+    X1, Y1, X2, Y2, _ = other
+    h_overlaps = (x1 <= X2) and (x2 >= X1)
+    v_overlaps = (y1 <= Y2) and (y2 >= Y1)
+    return h_overlaps and v_overlaps
+
+
+def has_overlap_with_group(det, group):
+    for other in group:
+        if has_overlap(det, other):
+            return True
+    return False
+
+
+def group_det_by_overlap(dets):
+    groups = [[]]
+    ordered_dets = sorted(dets, key=lambda x: (x[0], x[1]))
+    # add to groups the det that have overlap with others
+    for det in ordered_dets:
+        if not groups[-1]:
+            groups[-1].append(det)
+            continue
+        if has_overlap_with_group(det, groups[-1]):
+            groups[-1].append(det)
+        else:
+            groups.append([det])
```

Comment:
Since the first element is clearly a special case, maybe the code would be clearer by making this explicit?
```suggestion
    ordered_dets = sorted(dets, key=lambda x: (x[0], x[1]))
    groups = [ordered_dets[0]]
    # add to groups the det that have overlap with others
    for det in ordered_dets[1:]:
        if has_overlap_with_group(det, groups[-1]):
            groups[-1].append(det)
        else:
            groups.append([det])
```